rmw_opensplice_cpp/src/functions.cpp

```diff
@@ -180,16 +180,23 @@ rmw_create_publisher(
     return nullptr;
   }
 
-  DDS::DataWriterQos default_datawriter_qos;
-  status = dds_publisher->get_default_datawriter_qos(default_datawriter_qos);
+  DDS::DataWriterQos datawriter_qos;
+  status = dds_publisher->get_default_datawriter_qos(datawriter_qos);
   if (status != DDS::RETCODE_OK) {
     rmw_set_error_string("failed to get default datawriter qos");
     // printf("get_default_datawriter_qos() failed. Status = %d\n", status);
     return nullptr;
   }
 
+  // ensure the history depth is at least the requested queue size
+  if (datawriter_qos.history.kind == DDS::KEEP_LAST_HISTORY_QOS &&
```

Comment:
Always setting the history kind and depth here would prevent using vendor specific configuration options completely.

The history kind only support *ALL* and *DEPTH N*. If the default has been set to *ALL* the function doesn't reduce the history. If it is set to a specific *DEPTH* it ensures that it has at least the requested queue size. But if the default has been changed to a larger value it is not reduced.

Imo if the user changes the default configuration he is doing that for a specific reason and I would not overwrite that choice in our code.